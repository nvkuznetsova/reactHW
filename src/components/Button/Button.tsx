import React from 'react';

import './Button.styles.css';

export type ButtonProps = {
  btnLabel: string,
  btnClassName: string,
  disabled: boolean,
  click: () => void,
};

export const Button: React.FC<ButtonProps> = ({ 
  btnLabel,
  btnClassName,
  disabled,
  click,
}) => (
  <button
    className={`btn ${btnClassName}`}
    disabled={disabled}
    onClick={click}
    >
      {btnLabel}
    </button>
);