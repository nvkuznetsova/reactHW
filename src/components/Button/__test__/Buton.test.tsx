import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { Button, ButtonProps } from '../Button';

describe('Button', () => {
  const getProps = (): ButtonProps => ({
    btnLabel: 'Button',
    btnClassName: 'btn-class',
    disabled: false,
    click: () => null,
  });
  const setup = (props = {}): ShallowWrapper => (
    shallow(<Button {...getProps()} {...props}/>)
  );

  it('should match snapshot', () => {
    const component = setup();
    expect(component).toMatchSnapshot()
  });

  it('should match snapshot if button disabled', () => {
    const component = setup({ disabled: true });
    expect(component).toMatchSnapshot()
  });
});