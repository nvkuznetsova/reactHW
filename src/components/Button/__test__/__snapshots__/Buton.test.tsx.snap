// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  className="btn btn-class"
  disabled={false}
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Button should match snapshot if button disabled 1`] = `
<button
  className="btn btn-class"
  disabled={true}
  onClick={[Function]}
>
  Button
</button>
`;
